nemo/collections/nlp/data/language_modeling/megatron/megatron_batch_samplers.py

```diff
@@ -140,21 +140,25 @@ def __iter__(self):
 
 class MegatronPretrainingBatchSampler(BaseMegatronBatchSampler):
```

Comment:
The changes here will impact retro and nmt models too. Before megatron core, the batch would consist of `global_batch_size` number of samples.

After megatron core and the introduction of `dataloader_iter` to the `step` functions, we should pass microbatches and allow megatron to handle the fetching of batches itself.